### PR TITLE
Check Point Dome9 - fix fetch incidents

### PR DESCRIPTION
**Pull request by adi88d**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-19349)

## Description
- Fixed an issue where the integration fetching duplicate incidents.
- Changed the type of the *API key secret* and the *API key ID* parameters to `Encrypted`.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.yml [hunk 1]
```diff
@@ -9,13 +9,15 @@ configuration:
   required: true
   type: 0
 - name: api_key_id
-  display: API key ID
+  type: 9
```
Comment (DeanArbel) [inline Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.yml:12]: Good job prioritizing this, but please check with @mmhw on how to do this.
This change isn't as simple as just switching the param type.

File: Packs/CheckPointDome9/ReleaseNotes/1_0_3.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+#### Integrations
+##### Check Point Dome9 (CloudGuard)
+- Fixed an issue where the integration fetching duplicate incidents.
```
Comment (DeanArbel) [inline Packs/CheckPointDome9/ReleaseNotes/1_0_3.md:4]:
```suggestion
- Fixed an issue where the integration fetched duplicate incidents.
```

File: Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.py [hunk 3]
```diff
@@ -2120,8 +2127,21 @@ def main() -> None:
...
     base_url = params.get('base_url')
+
     key_id = params.get('api_key_id')
     key_secret = params.get('api_key_secret')
+    api_key_id_cred = params.get('api_key_id_cred', {}).get('password')
+    api_key_secret_cred = params.get('api_key_secret_cred', {}).get('password')
+
+    # get the api key and secret credentials
+    key_id = key_id or api_key_id_cred
+    key_secret = key_secret or api_key_secret_cred
```
Comment (mmhw) [inline Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.py:2138]:
```suggestion
    key_id = api_key_id_cred or key_id
    key_secret = api_key_secret_cred or key_secret
```

File: Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.yml [hunk 4]
```diff
@@ -10,12 +10,22 @@ configuration:
   type: 0
 - name: api_key_id
   display: API key ID
-  required: true
+  required: false
   type: 0
```
Comment (mmhw) [inline Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.yml:14]:
```suggestion
  type: 0
  hidden: true
```

File: Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.yml [hunk 5]
```diff
@@ -10,12 +10,22 @@ configuration:
...
 - name: api_key_secret
   display: API key secret
-  required: true
+  required: false
   type: 0
```
Comment (mmhw) [inline Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.yml:18]:
```suggestion
  type: 0
  hidden: true
```

**Review by mmhw — CHANGES_REQUESTED** [review]

File: Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.py (hunk 3 above)
Comment (mmhw) [reply Packs/CheckPointDome9/Integrations/CheckPointDome9/CheckPointDome9.py:2138]: To be able to update an existing instance with new credentials

**Review by mmhw — COMMENTED** [review]
For me it's good

**Review by DeanArbel — APPROVED** [review]

**Pull request merged**